No SessionId with ConsoleKit.
ConsoleKit exposes the session 'active' property in lowercase.

Index: js/gdm/loginDialog.js
--- js/gdm/loginDialog.js.orig
+++ js/gdm/loginDialog.js
@@ -52,6 +52,10 @@ const N_A11Y_MENU_COLUMNS = 2;
 
 Gio._promisify(Gio.File.prototype, 'load_contents_async');
 
+function haveSystemd() {
+    return GLib.access('/run/systemd/seats', 0) >= 0;
+}
+
 export const UserListItem = GObject.registerClass({
     Signals: {'activate': {}},
 }, class UserListItem extends St.Button {
@@ -1195,8 +1199,10 @@ export const LoginDialog = GObject.registerClass({
         const loginManager = LoginManager.getLoginManager();
         this._greeterSessionProxy = await loginManager.getCurrentSessionProxy();
         this._greeterSessionProxy?.connectObject('g-properties-changed', (proxy, properties) => {
-            const activeChanged = !!properties.lookup_value('Active', null);
-            if (activeChanged && proxy.Active)
+            // logind exposes 'Active', ConsoleKit exposes 'active'
+            const activeChanged = !!properties.lookup_value('Active', null) ||
+                                  !!properties.lookup_value('active', null);
+            if (activeChanged && (proxy.Active ?? proxy.active))
                 this._loginScreenSessionActivated();
         }, this);
     }
@@ -1266,6 +1272,7 @@ export const LoginDialog = GObject.registerClass({
     }
 
     async _findConflictingSession(startingSessionId) {
+        if (haveSystemd()) {
         const loginManager = LoginManager.getLoginManager();
         const sessions = await loginManager.listSessions();
         const [, , startingSessionOwner, ,] = sessions.find(([id, , , ,]) => id === startingSessionId);
@@ -1285,6 +1292,7 @@ export const LoginDialog = GObject.registerClass({
                 continue;
 
             return sessionProxy;
+        }
         }
 
         return null;
